src/main/java/net/javadiscord/javabot/listener/HugListener.java

```diff
@@ -24,6 +27,33 @@ public class HugListener extends ListenerAdapter {
 	private final AutoMod autoMod;
 	private final BotConfig botConfig;
 
+	private static final Pattern FUCKER = Pattern.compile("(fuck)(ing|er|k+)?", Pattern.CASE_INSENSITIVE);
+
+	private static String processFuck(String str) {
+		return FUCKER.matcher(str).replaceAll(matchResult -> {
+			String theFuck = matchResult.group(1);
+			String suffix = Objects.requireNonNullElse(matchResult.group(2), "");
+			String processedSuffix = switch(suffix.toLowerCase()) {
+				case "er", "ing" -> copyCase(suffix, 0, 'g') + suffix;
+				default -> suffix.toLowerCase().startsWith("k") ? copyCase(suffix, "g".repeat(suffix.length())) : "";
+			};
+			return processHug(theFuck) + processedSuffix;
+		});
+	}
+
+	private static String copyCase(String source, String toChange) {
```

Comment:
Wouldn't it make more sense to create a method that repeats a sequence of a character as often as present in some `String` with the correct case?
e.g. 
```java
private static String repeatWithSameCase(char toRepeat, String cases) {
	StringBuilder ret = new StringBuilder(cases.length());
	for (int i=0; i < cases.length; i++) {
		ret.append(copyCase(cases, i, toRepeat));
	}
	return ret.toString();
}
```

Reply:
that'd be oddly specific to make, i think just copying the case makes it a little clearer what actually happens

Reply:
I'm also OK with your approach but if you want to do that, please use better variable names than `a` and `b`.